#### Lab Assignment 1 - Jack Travers
#### Student ID: 200013389
#### GitHub URL: https://github.com/Jacktravers81/PY4SA23_Assignment

##### Section One: Python Basics

Task 1: Create an If...Else statement that will test whether a number is divisible by three.

In [79]:
def divisible (n):
    if n % 3 == 0:
        print (n, "is divisible by 3")
    else:
        print (n, "is not divisible by 3")
divisible (11)

11 is not divisible by 3


Task 2: Create an If...Else statement that will test whether a type of fruit, represented as a text string, is in a list of acceptable fruits (apple, orange, pear, kiwi, or strawberry).

In [43]:
fruit = ["apple", "orange", "pear", "kiwi", "strawberry"]

def library(check):
    if check in fruit:
        print (check, "is acceptable")
    else:
        print (check, "is not acceptable")

library("berry")

berry is not acceptable


Task 3: Create a function to calculate the distance between two coordinates using the haversine
formula. Write the following formula where the input parameters are a pair of coordinates as two
lists.

In [44]:
# COME BACK TO ME!!!!! 

a = sin²(φB - φA/2) + cos φA * cos φB * sin²(λB - λA/2)
c = 2 * atan2( √a, √(1−a) )
d = R ⋅ c

SyntaxError: invalid character '²' (U+00B2) (1688034982.py, line 5)

##### Section Two: Pandas and NumPy

Question 1: How many trees are of the Quercus or Acer genus?

In [11]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv", index_col=30)

trees_df.value_counts("Genus")

Genus
Pseudotsuga      6783
Acer             3586
Quercus          2089
Prunus           1333
Pinus             983
                 ... 
Eucommia            1
Eriobotrya          1
Daphniphyllum       1
Schima              1
Illicium            1
Length: 117, dtype: int64

Answers:
Genus *Acer*: 3586,
Genus *Quercus*: 2089

Question 2: : How many trees are of the Quercus or Acer genus and have a DBH larger than 50 inches?

In [55]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv")

just_acer = trees_df[(trees_df["Genus"]=="Acer") & (trees_df["DBH"]>50)][["Genus", "DBH"]]
print(just_acer)
print('')
# just_quercus = trees_df[(trees_df["Genus"]=="Quercus") & (trees_df["DBH"]>50)][["Genus", "DBH"]]
# print(just_quercus)
# print('')


      Genus   DBH
127    Acer  50.5
152    Acer  50.5
154    Acer  52.7
162    Acer  56.3
593    Acer  50.9
595    Acer  53.0
680    Acer  52.7
814    Acer  53.5
839    Acer  53.5
1936   Acer  51.6
3077   Acer  52.7
3116   Acer  59.2
3127   Acer  52.6
3129   Acer  50.3
3217   Acer  58.0
3232   Acer  53.3
3247   Acer  55.0
3251   Acer  58.0
3253   Acer  65.0
5423   Acer  65.9
6159   Acer  50.1
6409   Acer  73.0
8916   Acer  54.0
12329  Acer  60.0
12336  Acer  58.4
12338  Acer  61.1
12396  Acer  62.8
15847  Acer  55.1
17243  Acer  54.8
18122  Acer  54.3
19709  Acer  56.5
19894  Acer  54.0
20167  Acer  52.4
20224  Acer  56.3
20235  Acer  66.7
20655  Acer  53.7
23318  Acer  51.0
23481  Acer  59.4
23869  Acer  59.3
23892  Acer  63.7
23932  Acer  55.8
23957  Acer  59.5
25499  Acer  84.5
25506  Acer  63.0

